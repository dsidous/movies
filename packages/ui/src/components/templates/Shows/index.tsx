import React, { useState } from 'react';

import Filter from '../../organisms/Filter';
import Result from '../../molecules/Result';
import PageTransition from '../../atoms/PageTransition';
import { Genres } from '@typesRoots/movie';

interface Props {
  media: string;
  resultMedia: string;
  genres: Genres[];
}

const Shows: React.FC<Props> = ({ media, resultMedia, genres }) => {
  const [query, queryUpdate] = useState('');

  return (
    <PageTransition>
      <Filter
        genres={genres}
        queryUpdate={queryUpdate}
        media={media}
        query={query}
      >
        <Result query={query} resultMedia={resultMedia} />
      </Filter>
    </PageTransition>
  );
};

export default Shows;
