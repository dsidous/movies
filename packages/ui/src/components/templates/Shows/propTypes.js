import { object, array, bool } from 'prop-types';

export const contextTypes = {
  router: object.isRequired,
};

export const propTypes = {
  shows: array,
  genres: array,
  loading: bool,
  genresLoading: bool,
  configLoading: bool,
};

export const defaultProps = {};
