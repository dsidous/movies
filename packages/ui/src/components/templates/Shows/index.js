import React, { useState } from 'react';

import { propTypes } from './propTypes';
import Filter from '../../organisms/Filter';
import Result from '../../molecules/Result';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition';

const Shows = ({
  media,
  resultMedia,
  genres,
  loading,
  genresLoading,
  configLoading,
}) => {
  const [query, queryUpdate] = useState('');

  if (loading || genresLoading || configLoading) {
    return <Spinner />;
  }

  return (
    <PageTransition>
      <Filter genres={genres} queryUpdate={queryUpdate} media={media}>
        <Result query={query} resultMedia={resultMedia} />
      </Filter>
    </PageTransition>
  );
};

Shows.propTypes = propTypes;

export default Shows;
