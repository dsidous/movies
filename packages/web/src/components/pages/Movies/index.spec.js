import React from 'react';
import { shallow } from 'enzyme';
import Shows from '.';

const defaultProps = {
  location: { search: '1' },
};

describe('Pages/Movies', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Shows {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
