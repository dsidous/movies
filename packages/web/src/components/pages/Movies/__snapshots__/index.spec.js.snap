// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Movies should render as expected 1`] = `ShallowWrapper {}`;
